src/OpenLoco/Company.cpp

```diff
@@ -131,4 +133,65 @@ namespace OpenLoco
             unk.var_7C = totalRunCost;
         }
     }
+
+    // 0x004B8ED2
+    void Company::updateVehicleColours()
+    {
+        for (auto v : EntityManager::VehicleList())
+        {
+            if (v->owner != id())
+            {
+                continue;
+            }
+            Vehicles::Vehicle train(v);
+            for (auto& car : train.cars)
+            {
+                auto* vehObject = car.body->object();
+                auto colour = mainColours;
+                if (customVehicleColoursSet & (1 << vehObject->colour_type))
+                {
+                    colour = vehicleColours[vehObject->colour_type - 1];
+                }
+                for (auto& carComponent : car)
+                {
+                    carComponent.front->colour_scheme = colour;
+                    carComponent.back->colour_scheme = colour;
+                    carComponent.body->colour_scheme = colour;
+                }
+            }
+        }
+        Gfx::invalidateScreen();
+    }
+
+    // 0x0042F0C1
+    static void updateHeadquartersColourAtTile(const Map::TilePos2& pos, uint8_t zPos, Colour_t newColour)
+    {
+        auto tile = Map::TileManager::get(pos);
```

Comment:
Really we should be invalidating the screen at each of these tiles. ATM it doesn't matter as we invalidate the whole screen in the vehicle update colour function that is always called before this but it was just a thought i had.

Reply:
That's a good point. Currently, this function is only called by the ChangeCompanyColour game command, so it won't be a problem, as you said. Any preference either way?

Reply:
not really just an observation.